lighthouse-core/lib/asset-saver.js

```diff
@@ -272,11 +273,34 @@ async function logAssets(artifacts, audits) {
   });
 }
 
+/**
+ * @param {LH.DevtoolsLog} devtoolsLog
+ * @param {string} outputPath
+ * @return {Promise<void>}
+ */
+async function saveLanternNetworkData(devtoolsLog, outputPath) {
+  const context = /** @type {LH.Audit.Context} */ ({computedCache: new Map()});
+  const networkAnalysis = await NetworkAnalysisComputed.request(devtoolsLog, context);
+
+  /** @type {LH.PrecomputedLanternData} */
+  const lanternData = {additionalRttByOrigin: {}, serverResponseTimeByOrigin: {}};
+  for (const [origin, value] of networkAnalysis.additionalRttByOrigin.entries()) {
+    if (origin.startsWith('http')) lanternData.additionalRttByOrigin[origin] = value;
+  }
+
+  for (const [origin, value] of networkAnalysis.serverResponseTimeByOrigin.entries()) {
+    if (origin.startsWith('http')) lanternData.serverResponseTimeByOrigin[origin] = value;
+  }
```

Comment:
it feels like this shouldn't be in `asset-saver.js`, like it should be somewhere that can just be requested as `whatever.getSaveableLanternData()` or whatever, but I don't have a great suggestion. `Simulator`, since it's also in charge of understanding the format upon load?

Reply:
good point 👍 moved